src/js/node/events.js

```diff
@@ -247,15 +257,17 @@ EventEmitterPrototype.listeners = function listeners(type) {
   if (!events) return [];
   var handlers = events[type];
   if (!handlers) return [];
-  return handlers.map(x => x.listener ?? x);
+  // reverse array since handlers are stored in reverse internally
+  return handlers.reverse().map(x => x.listener ?? x);
```

Comment:
Use of `.map` and things tends to be slow when stacked on other ES5 array methods. Individually less so. But still keeping things bare bones may be better here.